File: swift/DesignPattern.playground/Pages/SimpleFactoryPattern.xcplaygroundpage/Sources/BeverageFactory.swift
import Foundation

open class BeverageFactory {
    public init() {}
    
    func createBeverage(beverageName: String) -> Beverage? {
        var beverage: Beverage?
        
        switch beverageName {
        case "black tea":
            beverage = BlackTea()
        case "green tea":
            beverage = GreenTea()
        default:
            break
        }
        
        return beverage
    }
}

File: kotlin/DesignPattern/src/main/kotlin/simplefactorypattern/source/BeverageFactory.kt
package simplefactorypattern.source

class BeverageFactory {
    fun createBeverage(beverageName: String): Beverage? {
        return when (beverageName) {
            "black tea" -> BlackTea()
            "green tea" -> GreenTea()
            else -> null
        }
    }
}